### ISSUE TITLE
Cleanup MODERNIZE_TESTS

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#853]
If MODERNIZE_TESTS is set to TRUE, cmake will convert scripts from classic mode to modern mode, using [gmtmodernize](https://github.com/GenericMappingTools/gmtmodernize).

However, gmtmodernize doesn't work as expected, and I think it's impossible to do the classic-modern conversion automatically. 

I think we should cleanup the codes related to MODERNIZE_TESTS.

**Comment by PaulWessel** [comment]
Yes, Leo and I in fact were talking about this a few hours ago.  The whole MODERNIZE stuff started 2 years ago when Leo first arrived and we imagined running gmtmodernize to auto-convert classic scripts on the fly to modern.  We later abandoned that but not until I had make changes to gmtest.in.  I will be doing those changes shortly; see #852 for context.

**Comment by joa-quim** [comment]
We still need a solution to -Jz in modern. One way of reducing its need (but not totally resolve the problem) would be to eliminate it use in several programs. I have once raised this issue (called it the _Flat programs issue__, probably in the old forum). For example _pscoast_ may require **-JZ** when the **-R** in history has all six _x_min/x_max/y_min/y_max/z_min/z_max_ values. Not using **-JZ** (or -Jz) results in an error saying that when -R...z_min/z_max, *-JZ* is mandatory. Relaxing the test that leads to this (would have to be done for the _flat modules_ only) would not only help this issue but also be  good in itself (classic mode included).

**Comment by PaulWessel** [comment]
If you can send me a script example that fails (ideally one that works and the same one that fails if -Jz is removed) and I can have a look for a solution.

**Comment by joa-quim** [comment]
```
echo 0 0 6 | psxyz -R-5/5/-1/6/0/10 -JM12c -JZ5c -Sc0.2c -K > lixo.ps

pscoast -R -Dc -J -W -O >> lixo.ps
pscoast [ERROR]: -R with six parameters requires -Jz|Z
pscoast [ERROR]: Syntax error -R option. Correct syntax:
        -R<xmin>/<xmax>/<ymin>/<ymax>[/<zmin>/<zmax>]
          Append r if giving lower left and upper right coordinates
        -Rg or -Rd for global domain
        -R<grdfile> to take the domain from a grid file
pscoast [ERROR]: Offending option -R-5/5/-1/6/0/10
```
Add -JZ and it work

**Comment by PaulWessel** [comment]
One problem is this: I can detect that **-R** has six items and check if **-Jz** or **-JZ** were given.  If not, we could add the missing **-Jz** or **-JZ**.  I can look in the history to see if one is present.  But if both are present then I have no way of knowing which one.  So you could quietly get the wrong option.

**Comment by joa-quim** [comment]
My idea is that the _flaters_ never require -JZ|z in first place. They do because they share a common test that requires it when -R has six items. It would have to be a condition in the general test to let go the _flaters_

**Comment by PaulWessel** [comment]
So I understand: You mean the other way around: When -Jz or JZ are not given then ignore the fact that -R has 6 items since not used.

**Comment by joa-quim** [comment]
Hm, yes. That looks simpler to implement since it does not require a list of exceptions.

**Comment by seisman** [comment]
Closed by #858.
